src/test/java/org/snomed/snowstorm/core/data/services/ConceptServiceTest.java

```diff
@@ -1699,6 +1703,138 @@ private boolean waitUntil(Supplier<Boolean> supplier, int maxSecondsToWait) {
+
+		// Create international concept with two descriptions
+		Description fsnDesc = new Description(fsnTerm).setTypeId(FSN).setCaseSignificance(ci).setAcceptabilityMap(intPreferred)
+				.setLang("en").setLanguageCode("en").addLanguageRefsetMember(GB_EN_LANG_REFSET, Concepts.PREFERRED, true);
+		Description synonymDesc = 	new Description(synonymTerm).setTypeId(SYNONYM).setCaseSignificance(ci).setAcceptabilityMap(intPreferred)
+				.setLang("en").setLanguageCode("en").addLanguageRefsetMember(GB_EN_LANG_REFSET, Concepts.PREFERRED, true);
+
+		Concept concept = new Concept()
+				.addDescription(fsnDesc)
+				.addDescription(synonymDesc)
+				.addAxiom(new Relationship(ISA, parentConceptId));
+		concept = conceptService.create(concept, mainBranchPath);
+
+		// Add refset membership
+		Iterator<Description> descriptionIterator =	concept.getDescriptions().iterator();
+		while(descriptionIterator.hasNext()) {
+			Description d = descriptionIterator.next();
+			memberService.createMember(mainBranchPath,
+				new ReferenceSetMember(d.getModuleId(), refSetId, d.getDescriptionId()));
+		}
+
+		// Create version
+		CodeSystem codeSystem = codeSystemService.find("SNOMEDCT");
+		codeSystemService.createVersion(codeSystem, 20210131, "20210131");
+
+		// Make a descriptor inactive
+		Description desc = concept.getDescriptions().stream().filter(d -> d.getTerm().equals(synonymTerm)).findFirst().get();
+		desc.setActive(false);
+		desc.setInactivationIndicator(Concepts.inactivationIndicatorNames.get(DUPLICATE));
+		desc.setAssociationTargets(Maps.newHashMap("POSSIBLY_EQUIVALENT_TO", Sets.newHashSet(anotherDescriptionId)));
+		Concept transferredConcept = simulateRestTransfer(concept);
+		conceptService.update(transferredConcept, mainBranchPath);
+		concept = conceptService.find(concept.getConceptId(), mainBranchPath);
+
+		// Create next version
+		codeSystemService.createVersion(codeSystem, 20210331, "20210331");
+
+		// Make the concept inactive
+		concept.setActive(false);
+		concept.setInactivationIndicator(Concepts.inactivationIndicatorNames.get(Concepts.DUPLICATE));
+		desc.setAssociationTargets(Maps.newHashMap("POSSIBLY_EQUIVALENT_TO", Sets.newHashSet(anotherConceptId)));
+		concept.setInactivationIndicator(Concepts.inactivationIndicatorNames.get(DUPLICATE));
+		transferredConcept = simulateRestTransfer(concept);
+		conceptService.update(transferredConcept, mainBranchPath);
+		concept = conceptService.find(concept.getConceptId(), mainBranchPath);
+
+		// Create next version
+		codeSystemService.createVersion(codeSystem, 20210531, "20210531");
+
+		// Create an extension code system (Sweden)
+		codeSystemService.createCodeSystem(new CodeSystem("SNOMEDCT-SE", extensionBranchPath));
+		branchService.updateMetadata(extensionBranchPath, ImmutableMap.of(Config.DEFAULT_MODULE_ID_KEY, extensionModule, Config.DEFAULT_NAMESPACE_KEY, "1000052"));
+		concept = conceptService.find(concept.getConceptId(), extensionBranchPath);
+
+		assertFalse(concept.isActive());
```

Comment:
Now please _reactivate_ the concept before adding the Swedish description. 

The test here is to see what happens to the _inactive_ inactivation indicators and historical associations.   So we need an active concept for that to happen (and then version again)